tesseract-server/src/handlers/logic_layer/geoadjacents.rs

```diff
@@ -0,0 +1,351 @@
+use tesseract_core::schema::{Cube, DimensionType};
+use crate::app::AppState;
+use crate::errors::ServerError;
+use crate::logic_layer::{LogicLayerConfig, CubeCache, Time};
+use super::super::util;
+use crate::handlers::logic_layer::{query_geoservice, GeoserviceQuery};
+
+
+/// Handles default aggregation when a format is not specified.
+/// Default format is jsonrecords.
+pub fn logic_layer_default_geoadjacents_handler(
+    (req, _cube): (HttpRequest<AppState>, Path<()>)
+) -> ActixResult<HttpResponse>
+{
+    logic_layer_geoadjacent(req, "jsonrecords".to_owned())
+}
+
+
+/// Handles aggregation when a format is specified.
+pub fn logic_layer_geoadjacents_handler(
+    (req, cube_format): (HttpRequest<AppState>, Path<(String)>)
+) -> ActixResult<HttpResponse>
+{
+    logic_layer_geoadjacent(req, cube_format.to_owned())
+}
+
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct LogicLayerQueryOpt {
+    pub cube: String,
+    #[serde(flatten)]
+    pub cuts: Option<HashMap<String, String>>,
+    parents: Option<bool>,
+    debug: Option<bool>,
+    exclude_default_members: Option<bool>,
+    locale: Option<String>,
+    sparse: Option<bool>,
+}
+
+
+pub fn logic_layer_geoadjacent(
+    req: HttpRequest<AppState>,
+    format: String,
+) -> ActixResult<HttpResponse>
+{
+    let format = format.parse::<FormatType>();
+    let format = match format {
+        Ok(f) => f,
+        Err(err) =>return Ok(HttpResponse::NotFound().json(err.to_string())),
+    };
+
+    info!("Format: {:?}", format);
+
+    let query = req.query_string();
+    let schema = req.state().schema.read().unwrap();
+    let debug = req.state().debug;
+
+    lazy_static!{
+        static ref QS_NON_STRICT: qs::Config = qs::Config::new(5, false);
+    }
+
+    let agg_query = match QS_NON_STRICT.deserialize_str::<LogicLayerQueryOpt>(query) {
+        Ok(q) => q,
+        Err(err) => return Ok(HttpResponse::NotFound().json(err.to_string()))
+    };
+
+    let logic_layer_config: Option<LogicLayerConfig> = match &req.state().logic_layer_config {
+        Some(llc) => Some(llc.read().unwrap().clone()),
+        None => None
+    };
+
+    let cube_name = match logic_layer_config.clone() {
+        Some(llc) => {
+            match llc.substitute_cube_name(agg_query.cube.clone()) {
+                Ok(cn) => cn,
+                Err(_) => agg_query.cube.clone()
+            }
+        },
+        None => agg_query.cube.clone()
+    };
+
+    let cube = match schema.get_cube_by_name(&cube_name) {
+        Ok(c) => c,
+        Err(err) => return Ok(HttpResponse::NotFound().json(err.to_string()))
+    };
+
+    let cube_cache = match req.state().cache.read().unwrap().find_cube_info(&cube_name) {
+        Some(cube_cache) => cube_cache,
+        None => return Ok(HttpResponse::NotFound().json("Unable to access cube cache".to_string()))
+    };
+
+    let cuts_map = match agg_query.cuts{
+        Some(cut_map) => cut_map,
+        None => HashMap::new(),
+    };
```

Comment:
Instead of creating an empty HashMap here, we should return an error if no cuts are provided.